homeassistant/components/media_player/ue_smart_radio.py

```diff
@@ -0,0 +1,217 @@
+"""
+Support for Logitech UE Smart Radios.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.ue_smart_radio/
+"""
+
+import logging
+from json import dumps
+import voluptuous as vol
+import requests
+
+from homeassistant.components.media_player import (
+    MediaPlayerDevice, MEDIA_TYPE_MUSIC, PLATFORM_SCHEMA,
+    SUPPORT_PLAY, SUPPORT_PAUSE, SUPPORT_STOP, SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_NEXT_TRACK, SUPPORT_TURN_ON, SUPPORT_TURN_OFF, SUPPORT_VOLUME_SET,
+    SUPPORT_VOLUME_MUTE)
+from homeassistant.const import (
+    CONF_EMAIL, CONF_PASSWORD, STATE_OFF, STATE_IDLE, STATE_PLAYING,
+    STATE_PAUSED, STATE_UNKNOWN)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+ICON = "mdi:radio"
+
+SUPPORT_UE_SMART_RADIO = SUPPORT_PLAY | SUPPORT_PAUSE | SUPPORT_STOP | \
+    SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | SUPPORT_TURN_ON | \
+    SUPPORT_TURN_OFF | SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_EMAIL): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Logitech UE Smart Radio platform."""
+    email = config.get(CONF_EMAIL)
+    password = config.get(CONF_PASSWORD)
+
+    session_request = requests.post("https://www.uesmartradio.com/user/login",
+                                    data={"email": email, "password":
+                                          password})
+    session = session_request.cookies["sdi_squeezenetwork_session"]
+
+    player_request = requests.post("http://decibel.logitechmusic.com/jsonrpc \
+                                   .js", cookies={"sdi_squeezenetwork_session":
+                                                  session},
+                                   data="{\"params\":[\"\", \
```

Comment:
Do you really need to send an escaped string as data? Can't you construct a `dict` of JSON form and use the `json` keyword argument in `requests.post`? http://docs.python-requests.org/en/master/user/quickstart/#more-complicated-post-requests

Reply:
An empty string or a null value needs to be sent for it to be accepted as that's usually where you specify the MAC address of the player you want. However, as we don't know the address of the player yet, you send a blank string instead.

Reply:
I didn't mean the empty string specifically. I mean that you could construct the data to send using a `dict`, which is converted to JSON. You could try using the `json` keyword to automatically encode the data, or if that doesn't work, the functions in the `json` module can automatically escape double quotes for you.

Reply:
Yep, I've done that now. Never knew you could use `json` before, very handy for use in the future, thanks for that. :cat2: